### PR TITLE
Ruby: Postupdate notes for assignment expressions.

### PR DESCRIPTION
**Pull request by michaelnebel**
This is a follow up of https://github.com/github/codeql/pull/10444
There is flow from "taint" to the field `field` on post-update node for `x` in the call to `taint_field`.

```ruby
def taint_field x
  x.field = "taint"
end

taint_field(y = x)
sink(x.field)
```

### CODE REVIEW

File: ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPrivate.qll [hunk 1]
```diff
@@ -52,6 +52,8 @@ private CfgNodes::ExprCfgNode getALastEvalNode(CfgNodes::ExprCfgNode n) {
   or
   result = n.(CfgNodes::ExprNodes::ConditionalExprCfgNode).getBranch(_)
   or
+  result = n.(CfgNodes::ExprNodes::AssignExprCfgNode).getRhs()
```
Comment (hvitved) [inline ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPrivate.qll:55]: This case should now be removed in `localFlowStepCommon`.

**Review by hvitved — CHANGES_REQUESTED** [review]

File: ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPrivate.qll (hunk 1 above)
Comment (michaelnebel) [reply ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPrivate.qll:55]: Oh - right!

**Review by hvitved — APPROVED** [review]
Thanks, LGTM. Let's see what DCA says.

**Comment by michaelnebel** [comment]
DCA doesn't show any change in alerts or any signs of performance regressions.

**Pull request merged**